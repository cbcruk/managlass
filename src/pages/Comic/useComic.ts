import { useParams } from 'react-router-dom'
import useSWR from 'swr'

function useComic() {
  const params = useParams<{ id: string }>()
  const {
    data,
    isValidating,
    error,
    mutate,
  } = useSWR(`/api/comic/${params.id}`, (url) =>
    fetch(url).then((r) => r.json())
  )

  return {
    data,
    error,
    mutate,
    isLoading: !data && !error,
    isValidating,
  }
}

export default useComic
