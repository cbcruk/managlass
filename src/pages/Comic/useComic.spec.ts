import { renderHook, act } from '@testing-library/react-hooks'
import Wrapper from './MockWrapper'
import useComic from './useComic'

test('useComic', async () => {
  const { result, waitForNextUpdate } = renderHook(() => useComic(), {
    wrapper: Wrapper,
  })

  await waitForNextUpdate()

  act(() => {
    result.current.mutate()
  })

  await waitForNextUpdate()

  expect(result.current).toMatchSnapshot()
})
