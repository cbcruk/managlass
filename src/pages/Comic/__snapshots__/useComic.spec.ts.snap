// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useComic 1`] = `
Object {
  "data": Object {
    "bookmark": "(1/1)",
    "chapters": Array [
      "6344349",
    ],
    "imageList": Array [
      "GZPmplOIna3h.jpeg",
      "_ZodNy4isTeK.jpeg",
      "ZkBnJQHcD0t5.jpeg",
      "A8Zin0lBRsaE.jpeg",
    ],
    "title": "1í™”",
  },
  "error": undefined,
  "isLoading": false,
  "isValidating": false,
  "mutate": [Function],
}
`;
